azure/datalake/store/core.py

```diff
@@ -192,6 +197,154 @@ def chmod(self, path, mod):
+        Sets the Access Control List (ACL) for a file or folder.
+
+        Note: this is not recursive, and applies only to the file or folder specified.
+
+        Parameters
+        ----------
+        path: str
+            Location to set the ACL on.
+        acl_spec: str
+            The ACL specification to set on the path in the format 
+            '[default:]user|group|other:[entity id or UPN]:r|-w|-x|-,[default:]user|group|other:[entity id or UPN]:r|-w|-x|-,...'
+        """
+        parms = {}
+        path = AzureDLPath(path).trim()
+        parms['aclSpec'] = acl_spec
+
+        self.azure.call('SETACL', path.as_posix(), **parms)
+
+    def modify_acl_entries(self, path, acl_spec):
+        """
+        Modifies existing Access Control List (ACL) entries on a file or folder.
+        If the entry does not exist it is added, otherwise it is updated based on the spec passed in.
+        No entries are removed by this process (unlike set_acl).
+
+        Note: this is not recursive, and applies only to the file or folder specified.
+
+        Parameters
+        ----------
+        path: str
+            Location to set the ACL entries on.
+        acl_spec: str
+            The ACL specification to use in modifying the ACL at the path in the format 
+            '[default:]user|group|other:[entity id or UPN]:r|-w|-x|-,[default:]user|group|other:[entity id or UPN]:r|-w|-x|-,...'
+        """
+        parms = {}
+        path = AzureDLPath(path).trim()
+        parms['aclSpec'] = acl_spec
+
+        self.azure.call('MODIFYACLENTRIES', path.as_posix(), **parms)
+
+    def remove_acl_entries(self, path, acl_spec):
+        """
+        Removes existing, named, Access Control List (ACL) entries on a file or folder.
+        If the entry does not exist already it is ignored.
+        Default entries cannot be removed this way, please use remove_default_acl for that.
+        Unnamed entries cannot be removed in this way, please use remove_acl for that. 
+
+        Note: this is not recursive, and applies only to the file or folder specified.
+
+        Parameters
+        ----------
+        path: str
+            Location to remove the ACL entries.
+        acl_spec: str
+            The ACL specification to remove from the ACL at the path in the format 
+            '[default:]user|group|other:[entity id or UPN]:r|-w|-x|-,[default:]user|group|other:[entity id or UPN]:r|-w|-x|-,...'
+        """
+        parms = {}
+        path = AzureDLPath(path).trim()
+        parms['aclSpec'] = acl_spec
+
+        self.azure.call('REMOVEACLENTRIES', path.as_posix(), **parms)
+    
+    def get_acl_status(self, path):
+        """
+        Gets Access Control List (ACL) entries for the specified file or directory.
+
+        Parameters
+        ----------
+        path: str
+            Location to get the ACL.
+        """
+        path = AzureDLPath(path).trim()
+
+        return self.azure.call('MSGETACLSTATUS', path.as_posix())['AclStatus']
```

Comment:
I count 3 uses of this pattern (5 if you count the two below where `parm` wasn't needed). It might be worth factoring this out to a helper function to cut down on repetitiveness, e.g.:
```python
def _acl_call(self, action, path, **kwargs):
    path = AzureDLPath(path).trim()
    self.azure.call(action, path.as_posix(), **kwargs)
```

Then all 5 instances become something like `self._acl_call('REMOVEACLENTRIES', path, aclSpec=acl_spec)`.